include/libtorrent/alert_types.hpp

```diff
@@ -241,12 +241,12 @@ TORRENT_VERSION_NAMESPACE_2
 	{
 		// internal
 		torrent_removed_alert(aux::stack_allocator& alloc
-			, torrent_handle const& h, sha1_hash const& ih);
+			, torrent_handle const& h, info_hash_t const& ih);
 
 		TORRENT_DEFINE_ALERT_PRIO(torrent_removed_alert, 4, alert_priority_critical)
 		static constexpr alert_category_t static_category = alert::status_notification;
 		std::string message() const override;
-		sha1_hash info_hash;
+		info_hash_t info_hash;
```

Comment:
I wonder if it would be worth it to improve backwards compatibility here, by deprecating the `info_hash` member and adding the `info_hash_t` member under a different name.

`info_hash_t` *is* implicitly convertible to a `sha1_hash`, so in some cases the current patch would just work.